ui/components-react/src/components-react/filter-builder/FilterBuilderState.ts

```diff
@@ -91,14 +91,11 @@ export class PropertyFilterBuilderActions {
   }
 
   /** Adds new rule or group of rules to the group specified by path. */
-  public addItem(path: string[], itemType: "RULE_GROUP" | "RULE") {
+  public addItem(path: string[]) {
```

Comment:
I don't think there is any reason to remove nesting support from PropertyFilterBuilder state and actions. It should be enough to change UI components to not use nesting. That way same state and actions implementation could be used for different UI components that do support nesting.